docs/contribute/cumulative-docs/content-patterns.md

```diff
@@ -0,0 +1,383 @@
+---
+navigation_title: "Example scenarios"
+---
+
+# Example scenarios
+
+## Code block content varies [code-block]
+
+Often the content in a code block will vary between situations (versions, deployment types, etc).
+There are a couple possible solutions.
+
+### Solution A: Use a code callout [code-block-callout]
+
+Using a code callout is the lightest-touch solution, but might not be sufficient in all cases.
+
+**When to use a code callout**:
+
+* The code block and its callouts fit vertically on a typical laptop screen.
+  This will reduce the risk of users copying the code snippet without reading the information in the callout.
+* Syntax is either just added or just removed — syntax is not modified.
+  It is difficult to communicate that some syntax is needed in more than one situation but varies depending on the situation.
+* The code block will not require more than 3 `applies_to`-related callouts.
+  At that point, the code becomes more difficult to read and use.
+
+**Best practices**:
+
+* Place the badge at the beginning of the callout.
+
+**Example**: A new option was made available in 9.1.0.
+
+::::{image} ./images/example-code-block-callout.png
+:screenshot:
+:alt:
+::::
+
+### Solution B: Use tabs [code-block-tabs]
+
+**When to use tabs**: If using a [code callout](#code-block-callout) isn't appropriate.
+
+**Best practices**:
+
+* Try to minimize the number of tabs where possible,
+  but do not mix tabs and `applies_to`-related code callouts.
+* Try not to include information surrounding a code block in the tabs.
+  Make the tab content as small as possible apart from the procedure itself.
+
+**Example**:
+
+::::{image} ./images/example-code-block-tabs.png
+:screenshot:
+:alt:
+::::
+
+## Workflows vary [workflow]
+
+When one or more steps in a process differs.
+
+### Solution A: Use inline `applies_to` [workflow-inline]
+
+Using inline `applies_to` badges to a few line items in an ordered list is the lightest-touch solution,
+but might not be sufficient in all cases.
+
+**When to use inline `applies_to`**:
+
+* Workflow steps that vary between situations can be easily isolated.
+* Each step that varies, only varies between 3 or fewer situations (deployment types, versions, etc).
+* There are no more than 3 steps that need to be split into multiple lines with `applies_to` badges.
+
+**Best practices**:
+
+* Follow the [best practices for ordered and unordered lists](/contribute/cumulative-docs/best-practices.md#ordered-and-unordered-lists)
+  including the order of items and the placement of labels.
+
+**Example**: Only one item in an ordered list varies between Serverless and stateful.
+
+::::{image} ./images/workflow-inline.png
+:screenshot:
+:alt:
+::::
+
+### Solution B: Use tabs [workflow-tabs]
```

Comment:
Isn't this guidance in conflict with earlier guidance:

> ### Tabs
> In the future ([elastic/docs-builder#1436](https://github.com/elastic/docs-builder/issues/1436)), tabs will be able to accept `applies_to` information. Until then, if an `applies_to` badge is relevant to the entire tab item, add the badge to the beginning of the content.

I think what you're suggesting here is that the tabs _should_ be labeled with the relevant version number or deployment method?

Then again, that's probably what we have to do until https://github.com/elastic/docs-builder/issues/1436 lands.